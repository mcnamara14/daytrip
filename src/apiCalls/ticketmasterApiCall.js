import { ticketmasterApiKey } from './apiKeys/ticketmasterApiKey';

export const ticketmasterApiCallRecentEvents = async (city, state, timeNow, timeIn2Days) => {
  const url = `https://app.ticketmaster.com/discovery/v2/events.json?startDateTime=${timeNow}&endDateTime=${timeIn2Days}&sort=date,name,asc&city=${city}&stateCode=${state}&apikey=${ticketmasterApiKey}`;

  try {
    const response = await fetch(url);
    const data = await response.json();
    const events = data._embedded.events;

    return events
  } catch (error) {
      alert(error)
  }
}





