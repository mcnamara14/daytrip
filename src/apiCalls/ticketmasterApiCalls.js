import { ticketmasterApiKey } from './apiKeys/ticketmasterApiKey';
import { 
  cleanRecentEventsSearch, 
  cleanRecentEvents 
} from '../dataCleaners/index';

export const fetchRecentEvents = async (city, state, timeNow) => {
  const prefix = 'https://app.ticketmaster.com/discovery/v2/events.json?';
  const start = `startDateTime=${timeNow}`;
  const sort = 'sort=date,name,asc';
  const userCity = `city=${city}`;
  const userState = `stateCode=${state}`;
  const apiKey = `apikey=${ticketmasterApiKey}`;

  const url = `${prefix}&${start}&${sort}&${userCity}&${userState}&${apiKey}`;

  try {
    const response = await fetch(url);
    const data = await response.json();
    const events = data._embedded.events;
    console.log(events)

    return events;
  } catch (error) {
    alert(error);
  }
};

export const fetchRecentEventsOnSearch = async (
  userCity, 
  userState, 
  timeNow, 
  keyword) => {
  const prefix = 'https://app.ticketmaster.com/discovery/v2/events.json?';
  const start = `startDateTime=${timeNow}`;
  const sort = 'sort=date,name,asc';
  const city = `city=${userCity}`;
  const state = `stateCode=${userState}`;
  const input = `keyword=${keyword}`;
  const key = `apikey=${ticketmasterApiKey}`;

  const url = `${prefix}${start}&${sort}&${city}&${state}&${input}&${key}`;

  try {
    const response = await fetch(url);
    const data = await response.json();
    const events = data._embedded.events;
  
    return cleanRecentEventsSearch(events);
  } catch (error) {
    console.log(error)
  }

};

export const fetchSelectedEvent = async (id) => {
  const prefix = `https://app.ticketmaster.com/discovery/v2/events/${id}.json?`;
  const url = `${prefix}apikey=${ticketmasterApiKey}`;

  try {
    const response = await fetch(url);
    const event = await response.json();
    
    return cleanRecentEvents([event]);
  } catch (error) {
    console.log('error');
  }
};










