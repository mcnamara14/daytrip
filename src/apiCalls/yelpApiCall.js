import { yelpApiKey } from './apiKeys/yelpApiKey';
import * as cleaners from '../dataCleaners/index';
import { suggestedRestaurantsCleaner } from '../dataCleaners/suggestedRestaurantsCleaner';

export const yelpApiCall = async () => {
  let headers = new Headers();
  headers.append("Authorization", "Bearer " + yelpApiKey);

  const result = await fetch("https://cors-anywhere.herokuapp.com/https://api.yelp.com/v3/businesses/search?term=delis&latitude=37.786882&longitude=-122.399972", {
    headers
  });
  const data = await result.json();
};

export const yelpFetchRestaurants = async (latitude, longitude, price, category) => {
  let headers = new Headers();
  headers.append("Authorization", "Bearer " + yelpApiKey);
  
  const result = await fetch(`https://cors-anywhere.herokuapp.com/https://api.yelp.com/v3/businesses/search?categories=${category}&latitude=${latitude}&longitude=${longitude}&price=${price}&radius=2500&sort_by=rating`, {
    headers
  });
  const data = await result.json();
  const restaurants = data.businesses;

  return suggestedRestaurantsCleaner(restaurants)
};