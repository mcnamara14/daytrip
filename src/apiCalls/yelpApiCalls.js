import { yelpApiKey } from './apiKeys/yelpApiKey';
import { 
  suggestedRestaurantsCleaner 
} from '../dataCleaners/suggestedRestaurantsCleaner';

export const fetchRestaurantsAndBars = async (
  latitude, 
  longitude, 
  price, 
  category) => {

  const corsAnywhereUrl = 'https://cors-anywhere.herokuapp.com/'
  const prefix = `${corsAnywhereUrl}https://api.yelp.com/v3/businesses/search?`;
  const cat = `categories=${category}`;
  const lat = `latitude=${latitude}`;
  const long = `longitude=${longitude}`;
  const cost = `price=${price}`;
  const radius = `radius=2500`;
  const sort = 'sort_by=rating';

  let headers = new Headers();
  headers.append("Authorization", "Bearer " + yelpApiKey);
  
  const result = 
    await fetch(`${prefix}${cat}&${lat}&${long}&${cost}&${radius}&${sort}`, 
      {
        headers
      });
  const data = await result.json();
  const restaurants = data.businesses;
  
  return suggestedRestaurantsCleaner(restaurants);
};