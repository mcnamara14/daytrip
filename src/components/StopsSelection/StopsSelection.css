.stopsSelectionContainer {
  background: #d9492c;
  width: calc(60% - 140px);
  margin-top: -100px;
  padding: 20px 70px;
  text-align: left;
}