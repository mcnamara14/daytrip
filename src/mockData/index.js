import { mockDirtyRecentEvents } from './mockRecentEvents';
import { mockCleanRestaurantAndBarData } from './mockRestaurantAndBarData';

export {
  mockDirtyRecentEvents,
  mockCleanRestaurantAndBarData
}