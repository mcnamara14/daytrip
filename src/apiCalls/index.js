import { ticketmasterApiCall } from './ticketmasterApiCall'; 
import { yelpApiCall } from './yelpApiCall'; 

export {
  ticketmasterApiCall,
  yelpApiCall
};