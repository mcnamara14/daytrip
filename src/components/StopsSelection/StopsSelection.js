import React, { Component } from 'react';
import { connect } from 'react-redux';
import './StopsSelection.css';
import StopSelection from '../../containers/StopSelection/StopSelection';
import { 
  storeSuggestedRestaurants, 
  storeSuggestedBars, 
  toggleFiltersError, 
  toggleEventError,
  toggleRestaurantBarError 
} from '../../actions';
import { fetchRestaurantsAndBars } from '../../apiCalls';
import PropTypes from 'prop-types';

export class StopsSelection extends Component {
  componentWillReceiveProps(nextProps) {
    if (nextProps.selectedEvent !== null && 
        nextProps.restaurantFilters.category !== undefined && 
        nextProps.barFilters.category !== undefined && 
        !this.props.filtersError &&
        !this.props.restaurantBarError) {
      setTimeout(() => {
        this.storeRestaurantsAndBars();
      }, 1000);
    }
  }

  storeRestaurantsAndBars = () => {
    const { 
      selectedEvent,
      restaurantFilters,
      barFilters
    } = this.props;

    if (restaurantFilters.category !== undefined && 
        barFilters.category !== undefined && 
        selectedEvent !== null) {
      const latitude = selectedEvent.latitude;
      const longitude = selectedEvent.longitude;
      this.storeSuggestedRestaurants(latitude, longitude);
      this.storeSuggestedBars(latitude, longitude);
    } else if (selectedEvent === null ) {
      this.toggleEventError();
    } else {
      this.toggleFiltersError();
    }
  }

  toggleEventError = () => {
    this.props.toggleEventError(true);
    setTimeout(() => {
      this.props.toggleEventError(false);
    }, 2000);
  }

  toggleFiltersError = () => {
    this.props.toggleFiltersError(true);
    setTimeout(() => {
      this.props.toggleFiltersError(false);
    }, 2000);
  }

  toggleRestaurantBarError = () => {
    this.props.toggleRestaurantBarError(true);
    setTimeout(() => {
      this.props.toggleRestaurantBarError(false);
    }, 2000);
  }

  storeSuggestedRestaurants = async (latitude, longitude) => {
    const {
      restaurantFilters,
      storeSuggestedRestaurants
    } = this.props;

    const price = restaurantFilters.priceRange.sort().join();
    const category = restaurantFilters.category;
    const suggestedRestaurants = 
      await fetchRestaurantsAndBars(latitude, longitude, price, category);
    suggestedRestaurants.length ? 
      storeSuggestedRestaurants(suggestedRestaurants) :
      this.toggleRestaurantBarError();
  }

  storeSuggestedBars = async (latitude, longitude) => {
    const {
      barFilters,
      storeSuggestedBars
    } = this.props;

    if (barFilters.priceRange.length) {
      const price = barFilters.priceRange.sort().join();
      const category = barFilters.category;
      const suggestedBars = 
        await fetchRestaurantsAndBars(latitude, longitude, price, category);
      storeSuggestedBars(suggestedBars);
    }

  }

  render() {
    return (
      <div className="stopsSelection">
        { this.props.filtersError ? 
          <p className="filtersError">
            Please add a price and category for each stop.
          </p> : 
          null }
        { this.props.restaurantBarError ? 
          <p className="filtersError">
          No restaurants or bars near by matching your filters. 
            <br/> Select new filters.</p> : 
          null }
        <div className="stopsSelectionContainer">
          <StopSelection type={'before'}/>
          <hr />
          <StopSelection type={'after'}/>
        </div>
        <button onClick={this.storeRestaurantsAndBars}>Add filters</button>
      </div>
    );
  }
}

StopsSelection.propTypes = {
  selectedEvent: PropTypes.object,
  restaurantFilters: PropTypes.object,
  barFilters: PropTypes.object,
  filtersError: PropTypes.func,
  restaurantBarError: PropTypes.func,
  toggleEventError: PropTypes.func,
  toggleFiltersError: PropTypes.func,
  toggleRestaurantBarError: PropTypes.func
};

export const mapDispatchToProps = (dispatch) => ({
  storeSuggestedRestaurants: (restaurants) => {
    return dispatch(storeSuggestedRestaurants(restaurants));
  },
  storeSuggestedBars: (bars) => dispatch(storeSuggestedBars(bars)),
  toggleFiltersError: (boolean) => dispatch(toggleFiltersError(boolean)),
  toggleEventError: (boolean) => dispatch(toggleEventError(boolean)),
  toggleRestaurantBarError: (boolean) => {
    return dispatch(toggleRestaurantBarError(boolean));
  }
});

export const mapStateToProps = (state) => ({
  selectedEvent: state.selectedEvent,
  restaurantFilters: state.restaurantFilters,
  barFilters: state.barFilters,
  filtersError: state.filtersError,
  restaurantBarError: state.restaurantBarError
});

export default connect(mapStateToProps, mapDispatchToProps)(StopsSelection);