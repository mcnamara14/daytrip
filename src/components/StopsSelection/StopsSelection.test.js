import React from 'react';
import ReactDOM from 'react-dom';
import { StopsSelection, mapDispatchToProps } from './StopsSelection';
import { shallow, mount } from 'enzyme';
import * as fetchCalls from '../../apiCalls';
jest.mock('../../apiCalls/yelpApiCalls');
import { mockCleanSuggestedRestaurant } from '../../mockData';

describe('stopsSelection', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<StopsSelection />);
  
    expect(wrapper).toMatchSnapshot();
  });

  describe('componentWillReceiveProps', () => {
    it('should call storeRestaurantsAndBars when there is an event and filters', () => {
      jest.useFakeTimers();
      const spy = jest.spyOn(StopsSelection.prototype, 'componentWillReceiveProps');
      const wrapper = mount(<StopsSelection 
        selectedEvent={{ title: 'T-Swift'}} 
        restaurantFilters={{category:'newamerican'}}
        barFilters={{category:'newamerican'}}
        filtersError={false}
        restaurantBarError={false}
      />);

      wrapper.instance().storeRestaurantsAndBars = jest.fn();
    
      expect(wrapper.instance().storeRestaurantsAndBars).not.toHaveBeenCalled();
    
      wrapper.setProps({
        selectedEvent: {
          title: 'Backstreet Boys'
        }
      });
    
      wrapper.update();
      jest.runAllTimers();
    
      expect(wrapper.instance().storeRestaurantsAndBars).toHaveBeenCalled();
    })

    it('should not call storeRestaurantsAndBars when an event or filters undefined or null', () => {
      jest.useFakeTimers();
      const spy = jest.spyOn(StopsSelection.prototype, 'componentWillReceiveProps');
      const wrapper = mount(<StopsSelection 
        selectedEvent={{}} 
        restaurantFilters={{category:'newamerican'}}
        barFilters={{category:'newamerican'}}
        filtersError={false}
        restaurantBarError={false}
      />);

      wrapper.instance().storeRestaurantsAndBars = jest.fn();
    
      expect(wrapper.instance().storeRestaurantsAndBars).not.toHaveBeenCalled();
    
      wrapper.setProps({
        selectedEvent: null
      });
    
      wrapper.update();
      jest.runAllTimers();
    
      expect(wrapper.instance().storeRestaurantsAndBars).not.toHaveBeenCalled();
    })
  });

  describe('storeRestaurantsAndBars', () => {
    let wrapper;
    let mockEvent;
    let mockRestaurantFilters;
    let mockBarFilters;

    beforeEach(() => {
      mockEvent = {
        latitude: '1234',
        longitude: '6789',
        title: 'T-Swift'
      }
      mockRestaurantFilters = {
        category: 'newamerican'
      }
      mockBarFilters = {
        category: 'divebar'
      }

      wrapper = shallow(<StopsSelection 
        selectedEvent={mockEvent} 
        restaurantFilters={mockRestaurantFilters} 
        barFilters={mockBarFilters} />);

      wrapper.instance().storeSuggestedRestaurants = jest.fn();
      wrapper.instance().storeSuggestedBars = jest.fn();
      wrapper.instance().toggleEventError = jest.fn();
      wrapper.instance().toggleFiltersError= jest.fn();
    })

    it('should call storeSuggestedRestaurants with the correct args when there are restaraunts and bars', () => {
      wrapper.instance().storeRestaurantsAndBars();

      const expected = ["1234", "6789"];

      expect(wrapper.instance().storeSuggestedRestaurants).toHaveBeenCalledWith(...expected)
    })

    it('should call storeSuggestedBars with the correct args when there are restaraunts and bars', () => {
      wrapper.instance().storeRestaurantsAndBars();

      const expected = ["1234", "6789"];

      expect(wrapper.instance().storeSuggestedBars).toHaveBeenCalledWith(...expected)
    })

    it('should call toggleEventError when no event is selected', () => {
      mockEvent = null

      wrapper = shallow(<StopsSelection 
        selectedEvent={mockEvent} 
        restaurantFilters={mockRestaurantFilters} 
        barFilters={mockBarFilters} />);

      wrapper.instance().storeSuggestedRestaurants = jest.fn();
      wrapper.instance().storeSuggestedBars = jest.fn();
      wrapper.instance().toggleEventError = jest.fn();
      wrapper.instance().toggleFiltersError= jest.fn();
      
      wrapper.instance().storeRestaurantsAndBars();

      const expected = ["1234", "6789"];

      expect(wrapper.instance().toggleEventError).toHaveBeenCalled()
    })

    it('should call toggleFiltersError when filters are undefined', () => {
      mockEvent = {};
      mockRestaurantFilters = {
        category: undefined
      }

      wrapper = shallow(<StopsSelection 
        selectedEvent={mockEvent} 
        restaurantFilters={mockRestaurantFilters} 
        barFilters={mockBarFilters} />);

      wrapper.instance().storeSuggestedRestaurants = jest.fn();
      wrapper.instance().storeSuggestedBars = jest.fn();
      wrapper.instance().toggleEventError = jest.fn();
      wrapper.instance().toggleFiltersError= jest.fn();
      
      wrapper.instance().storeRestaurantsAndBars();

      const expected = ["1234", "6789"];

      expect(wrapper.instance().toggleFiltersError).toHaveBeenCalled()
    })
  })

  describe('toggleLocation', () => {
    it('should call toggleLocation with correct argument then call again 2 seconds', () => {
      const wrapper = shallow(<StopsSelection toggleEventError={jest.fn()} />);
      const result = wrapper.instance().props.toggleEventError;

      jest.useFakeTimers();

      wrapper.instance().toggleEventError();

      expect(result).toHaveBeenCalledWith(true);

      jest.runAllTimers();

      expect(result).toHaveBeenCalledWith(false);    })
  })

  describe('toggleFiltersError', () => {
    it('should call toggleFiltersError with correct argument then call again 2 seconds', () => {
      const wrapper = shallow(<StopsSelection toggleFiltersError={jest.fn()} />);
      const result = wrapper.instance().props.toggleFiltersError;

      jest.useFakeTimers();

      wrapper.instance().toggleFiltersError();

      expect(result).toHaveBeenCalledWith(true);

      jest.runAllTimers();

      expect(result).toHaveBeenCalledWith(false);    })
  })

  describe('toggleRestaurantBarError', () => {
    it('should call toggleRestaurantBarError with correct argument then call again 2 seconds', () => {
      const wrapper = shallow(<StopsSelection toggleRestaurantBarError={jest.fn()} />);
      const result = wrapper.instance().props.toggleRestaurantBarError;

      jest.useFakeTimers();

      wrapper.instance().toggleRestaurantBarError();

      expect(result).toHaveBeenCalledWith(true);

      jest.runAllTimers();

      expect(result).toHaveBeenCalledWith(false);    })
  })

  describe('storeSuggestedRestaurants', () => {
    let wrapper;
    let mockRestaurantFilters;

    beforeEach(() => {
      mockRestaurantFilters = {
        category: 'newamerican',
        priceRange: ['1', '3']
      }

      fetchCalls.fetchRestaurantsAndBars.mockImplementation(() => mockCleanSuggestedRestaurant)

      wrapper = shallow(<StopsSelection 
        restaurantFilters={mockRestaurantFilters}
        storeSuggestedBars={jest.fn()}
        storeSuggestedRestaurants={jest.fn()} 
      />);
    })

    it('should call fetchRestaurantsAndBars with the correct args ', async () => {
      await wrapper.instance().storeSuggestedRestaurants("39.735001", "-105.020401");

      const expected = ["39.735001", "-105.020401", "1,3", "newamerican"];

      expect(fetchCalls.fetchRestaurantsAndBars).toHaveBeenCalledWith(...expected);
    });

    it('should call storeSuggestedRestaurants in props with the correct args when there are suggested restaurants', async () => {
      await wrapper.instance().storeSuggestedRestaurants("39.735001", "-105.020401");

      const expected = ["39.735001", "-105.020401", "1,3", "newamerican"];

      expect(wrapper.instance().props.storeSuggestedRestaurants).toHaveBeenCalledWith(mockCleanSuggestedRestaurant);
    });

    it('should call toggleRestaurantBarError when there are no suggested restaurants', async () => {
      fetchCalls.fetchRestaurantsAndBars.mockImplementation(() => [])

      wrapper = shallow(<StopsSelection 
        restaurantFilters={mockRestaurantFilters}
        storeSuggestedBars={jest.fn()}
        storeSuggestedRestaurants={jest.fn()}
        toggleRestaurantBarError={jest.fn()} 
      />);

      wrapper.instance().toggleRestaurantBarError = jest.fn();

      await wrapper.instance().storeSuggestedRestaurants("39.735001", "-105.020401");

      const expected = ["39.735001", "-105.020401", "1,3", "newamerican"];

      expect(wrapper.instance().toggleRestaurantBarError).toHaveBeenCalled();
    });
  });

  describe('storeSuggestedBars', () => {
    let wrapper;
    let mockFilters;

    beforeEach(() => {
      mockFilters= {
        category: 'newamerican',
        priceRange: ['1', '3']
      }

      fetchCalls.fetchRestaurantsAndBars.mockImplementation(() => mockCleanSuggestedRestaurant)

      wrapper = shallow(<StopsSelection 
        barFilters={mockFilters}
        storeSuggestedBars={jest.fn()}
        storeSuggestedRestaurants={jest.fn()} 
      />);
    })

    it('should call fetchRestaurantsAndBars with correct args when there are bar filters', async () => {
      await wrapper.instance().storeSuggestedBars();

      const expected = ["39.735001", "-105.020401", "1,3", "newamerican"];

      expect(fetchCalls.fetchRestaurantsAndBars).toHaveBeenCalledWith(...expected);
    })
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the correct params on storeSuggestedRestaurants', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = {
        type: 'STORE_SUGGESTED_RESTAURANTS',
        restaurants: mockCleanSuggestedRestaurant
      };

      mappedProps.storeSuggestedRestaurants(mockCleanSuggestedRestaurant);

      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });

    it('should call dispatch with the correct params on storeSuggestedBars', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = {
        type: 'STORE_SUGGESTED_BARS',
        bars: mockCleanSuggestedRestaurant
      };

      mappedProps.storeSuggestedBars(mockCleanSuggestedRestaurant);

      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });

    it('should call dispatch with the correct params on toggleFiltersError', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = {
        type: 'TOGGLE_FILTERS_ERROR',
        boolean: true
      };

      mappedProps.toggleFiltersError(true);

      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });

    it('should call dispatch with the correct params on toggleEventError', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = {
        type: 'TOGGLE_EVENT_ERROR',
        boolean: true
      };

      mappedProps.toggleEventError(true);

      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });

    it('should call dispatch with the correct params on toggleRestaurantBarError', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = {
        type: 'TOGGLE_RESTAURANT_BAR_ERROR',
        boolean: true
      };

      mappedProps.toggleRestaurantBarError(true);

      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });
  });

})

