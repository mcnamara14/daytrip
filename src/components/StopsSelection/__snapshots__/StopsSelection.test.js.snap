// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stopsSelection should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StopsSelection />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        null,
        <div
          className="stopsSelectionContainer"
>
          <StopSelection
                    type="before"
          />
          <hr />
          <StopSelection
                    type="after"
          />
</div>,
        <button
          onClick={[Function]}
>
          Add filters
</button>,
      ],
      "className": "stopsSelection",
    },
    "ref": null,
    "rendered": Array [
      null,
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <StopSelection
              type="before"
/>,
            <hr />,
            <StopSelection
              type="after"
/>,
          ],
          "className": "stopsSelectionContainer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "type": "before",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "type": "after",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add filters",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Add filters",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          null,
          <div
            className="stopsSelectionContainer"
>
            <StopSelection
                        type="before"
            />
            <hr />
            <StopSelection
                        type="after"
            />
</div>,
          <button
            onClick={[Function]}
>
            Add filters
</button>,
        ],
        "className": "stopsSelection",
      },
      "ref": null,
      "rendered": Array [
        null,
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <StopSelection
                type="before"
/>,
              <hr />,
              <StopSelection
                type="after"
/>,
            ],
            "className": "stopsSelectionContainer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "type": "before",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "type": "after",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add filters",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add filters",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
